Fix sign-compare warnings in tests: osquery/registry/tests/registry_tests.cpp:245

The following compiler warning is generated during most builds:
```
In file included from /home/teddy/git/osquery/osquery/registry/tests/registry_tests.cpp:11:
/home/teddy/git/osquery/third-party/gtest-1.7.0/include/gtest/gtest.h:1448:16: warning: comparison of integers of different signs: 'const unsigned long' and 'const int' [-Wsign-compare]
  if (expected == actual) {
      ~~~~~~~~ ^  ~~~~~~
/home/teddy/git/osquery/third-party/gtest-1.7.0/include/gtest/gtest.h:1484:12: note: in instantiation of function template specialization 'testing::internal::CmpHelperEQ<unsigned long, int>' requested here
    return CmpHelperEQ(expected_expression, actual_expression, expected,
           ^
/home/teddy/git/osquery/osquery/registry/tests/registry_tests.cpp:245:3: note: in instantiation of function template specialization 'testing::internal::EqHelper<false>::Compare<unsigned long, int>' requested here
  EXPECT_EQ(RegistryFactory::getModule(), 0);
  ^
/home/teddy/git/osquery/third-party/gtest-1.7.0/include/gtest/gtest.h:1979:53: note: expanded from macro 'EXPECT_EQ'
                      EqHelper<GTEST_IS_NULL_LITERAL_(expected)>::Compare, \
                                                                  ^
1 warning generated.
```

These warnings should be treated as errors, but we should at least fix this comparison.

Something like:
```cpp
EXPECT_EQ(0U, RegistryFactory::getModule());
```
will do the trick!

If you've never compiled/tested osquery before start here: http://osquery.readthedocs.io/en/stable/development/building/

--- a/osquery/registry/tests/registry_tests.cpp
+++ b/osquery/registry/tests/registry_tests.cpp
@@ -242,7 +242,7 @@ TEST_F(RegistryTests, test_registry_modules) {
   RegistryFactory::locked(false);
 
   // Test initializing a module load and the module's registry modifications.
-  EXPECT_EQ(RegistryFactory::getModule(), 0);
+  EXPECT_EQ(0U, RegistryFactory::getModule());
   RegistryFactory::initModule("/my/test/module");
   // The registry is locked, no modifications during module global ctors.
   EXPECT_TRUE(RegistryFactory::locked());
@@ -271,6 +271,6 @@ TEST_F(RegistryTests, test_registry_modules) {
   EXPECT_TRUE(RegistryFactory::locked());
   // And the registry is no longer using a module.
   EXPECT_FALSE(RegistryFactory::usingModule());
-  EXPECT_EQ(RegistryFactory::getModule(), 0);
+  EXPECT_EQ(0U, RegistryFactory::getModule());
 }
 }

